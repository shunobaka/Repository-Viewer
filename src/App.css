body {
  background: url('./images/background.jpg');
}

h2 {
  margin-top: 10px;
}

.green-text {
  color: #bada55;
}

.landing {
  position: fixed;
  bottom: 0;
  right: 0;
  width: 100%;
  height: 100vh;
  z-index: -1;
}

.landing-inner {
  color: #bada55;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.home-button {
  color: #bada55 !important;
}

button.btn-outline-success {
  color: #bada55;
  border-color: #bada55;
}
