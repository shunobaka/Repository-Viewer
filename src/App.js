import React, { Fragment } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

// Custom components
import Navigation from './components/layout/Navigation';
import Landing from './components/layout/Landing';
import Routes from './components/routes/Routes';

// Redux
import { Provider } from 'react-redux';
import store from './store';

// Styling
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

const App = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Fragment>
          <Navigation />
          <Switch>
            <Route exact path="/" component={Landing} />
            <Route component={Routes} />
          </Switch>
        </Fragment>
      </BrowserRouter>
    </Provider>
  );
};

export default App;
