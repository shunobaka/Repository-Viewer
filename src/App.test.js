/**
 * @fileoverview Defines snapshot unit tests for App component.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
