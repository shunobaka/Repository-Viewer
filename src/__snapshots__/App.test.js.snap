// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-dark"
  >
    <span
      className="navbar-brand"
    >
      <a
        className="home-button"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <form
        className="ml-auto form-inline"
        onSubmit={[Function]}
      >
        <input
          className="mr-sm-2 form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <a
          href="/search/"
          onClick={[Function]}
        >
          <button
            className="btn btn-outline-success"
            disabled={false}
            type="button"
          >
            Search
          </button>
        </a>
      </form>
    </div>
  </nav>,
  <section
    className="landing"
  >
    <div
      className="landing-inner"
    >
      <h1>
        Welcome to Github-Repo-Viewer
      </h1>
      <h3>
        Search for a user in the navigation above to view repositories.
      </h3>
    </div>
  </section>,
]
`;
